cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -877,16 +877,16 @@ private V1PodSpec createPodSpec(Location runtimeConfigLocation,
     RuntimeSpecification mainRuntimeSpec = getMainRuntimeSpecification(runtimeSpecs);
     environs.putAll(environments.get(mainRuntimeSpec.getName()));
 
-    containers.add(createContainer(mainRuntimeSpec.getName(), podInfo.getContainerImage(), workDir,
-                                   createResourceRequirements(mainRuntimeSpec.getResourceSpecification()),
+    containers.add(createContainer(mainRuntimeSpec.getName(), podInfo.getContainerImage(), podInfo.getImagePullPolicy(),
```

Comment:
what happens when the image pull policy is not explicitly provided for the parent pods? Does `podInfo.getImagePullPolicy()` would return default value or `null` in that case?

Reply:
I just did a test, and it looks like imagePullPolicy is [always set upon object creation](https://kubernetes.io/docs/concepts/containers/images/#updating-images):
```
The value of imagePullPolicy of the container is always set when the object is first created, and is not updated if the image's tag later changes.
```